waspc/src/Wasp/Analyzer/TypeChecker/Internal.hs

```diff
@@ -103,7 +103,7 @@ inferExprType (P.List values) = do
       return $ List (toList unifiedValues) (ListType unifiedType)
 -- Apply [Dict], and also check that there are no duplicate keys in the dictionary
 inferExprType (P.Dict entries) = do
-  typedEntries <- zip (map fst entries) <$> mapM (inferExprType . snd) entries
+  typedEntries <- mapM (\(k, v) -> (k,) <$> inferExprType v) entries
```

Comment:
This is old code, but as I was passing by I decided to refactor it a bit to make it more readable.